compiler/modules/front.syntax/token.bal

```diff
@@ -92,6 +92,16 @@ function toToken(string t) returns Token {
     return <Token>t;
 }
 
+type TokenizerState readonly & record {
+    int lineIndex;
+    int fragCodeIndex;
+    int codePointIndex;
+    int fragmentIndex;
+    int tokenStartCodePointIndex;
+    Mode mode;
+    Token? curTok;
```

Comment:
Another options is to have `Token curTok;` and change `TokenizerState tokState` to `TokenizerState? tokState` in `PreModulePart`'s `init`.

Reply:
What you have seems fine to me.